### ISSUE TITLE
improvements to `enableSAMLForDomain`

### ISSUE THREAD

**Issue by tiffanyhan** [ParabolInc/parabol#4383]
## Issue - Enhancement

For various reasons, setting up SAML still requires manual developer intervention. We should robustify the mutation so that non-technical folks can reliably set up sso for clients.

### Acceptance Criteria (optional)
`enableSAMLForDomain` mutation should:
- [ ] before setting `id` and `domain` fields, top level domain should be stripped out of input value if present
- [ ] `url` field should not normalize case, or should only normalize parts of url which are case insensitive i.e. query parameters should preserve case
- [ ] if the `url` value contains `"microsoftonline"`, we should add the deflate encoded `?SAMLRequest=` query param value to the `url` field.
- [ ] add `createdAt`, `updatedAt` and `deletedAt` ts fields
- [ ] check if valid xml for `metadata` field

- **Estimated effort:** 8 hours ([see CONTRIBUTING.md](https://github.com/ParabolInc/parabol/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by tiffanyhan** [comment]
```
const zlib = require('zlib')

/* hard coded values below should be dynamic */
const uuid = '002f199c-5311-4981-bb15-9fcf066c1454'
const client = 'effectory'
const destination = 'https://login.microsoftonline.com/e9792fd7-4044-47e7-a40d-3fba46f1cd09/saml2'

const msg = `
<samlp:AuthnRequest
    xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
    xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" ID="_${uuid}" Version="2.0" IssueInstant="${(new Date()).toISOString()}" Destination="${destination}" ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" AssertionConsumerServiceURL="https://action.parabol.co/saml/${client}">
<saml:Issuer>https://action.parabol.co/saml-metadata/${client}</saml:Issuer>
    <samlp:NameIDPolicy Format="urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress" AllowCreate="false"/>
</samlp:AuthnRequest>
`
const deflated = zlib.deflateRawSync(msg)
const b64encoded = deflated.toString('base64')
const URIencoded = encodeURIComponent(b64encoded)
console.log(URIencoded)
```

**Comment by mattkrick** [comment]
fixed in v5.29.0
